### ISSUE TITLE
composer-plugin-api 1.0.0 not found while installing extension (imagine)

### ISSUE THREAD

**Issue by nbdanton** [composer/composer#5422]
With the following composer.json:

```json
{
"minimum-stability": "stable",
    "require": {
        "php": ">=5.4.0",
        "yiisoft/yii2": ">=2.0.6",
        "yiisoft/yii2-bootstrap": "*",
        "yiisoft/yii2-swiftmailer": "*",
        "dmstr/yii2-adminlte-asset": "*",
        "kartik-v/yii2-grid": "@dev",
        "kartik-v/yii2-mpdf": "dev-master",
        "kartik-v/yii2-editable": "*",
        "kartik-v/yii2-widgets": "@dev",
        "devgroup/yii2-jsoneditor": "*",
        "kartik-v/yii2-date-range": "dev-master",
        "kartik-v/yii2-datecontrol": "dev-master",
        "yiisoft/yii2-twig": "^2.0",
        "2amigos/yii2-ckeditor-widget": "~1.0",
        "linslin/yii2-curl": "*",
        "smallbearsoft/yii2-ajax": "*",
		"yiisoft/yii2-imagine": "^2.0"
    },
    "require-dev": {
        "yiisoft/yii2-codeception": "*",
        "yiisoft/yii2-debug": "*",
        "yiisoft/yii2-gii": "^2.0",
        "yiisoft/yii2-faker": "*"
    },
    "config": {
        "process-timeout": 1800
    },
    "extra": {
        "asset-installer-paths": {
            "npm-asset-library": "vendor/npm",
            "bower-asset-library": "vendor/bower"
        }
    }
}
```

When I run this command:
```
php composer.phar require --prefer-dist yiisoft/yii2-imagine
```
I get this output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - yiisoft/yii2-composer 2.0.3 requires composer-plugin-api 1.0.0 -> no match
ing package found.
    - yiisoft/yii2-composer 2.0.3 requires composer-plugin-api 1.0.0 -> no match
ing package found.
    - yiisoft/yii2-composer 2.0.3 requires composer-plugin-api 1.0.0 -> no match
ing package found.
    - Installation request for yiisoft/yii2-composer (locked at 2.0.3) -> satisf
iable by yiisoft/yii2-composer[2.0.3].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your min
imum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more det
ails.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further commo
n problems.

Installation failed, reverting ./composer.json to its original content.
```

**Comment by Seldaek** [comment]
That's probably because you run composer v1.1.* in which the plugin api is 1.1.0 and that yii plugin is broken and doesn't require ^1.0 but 1.0.0

**Comment by nbdanton** [comment]
Thanks for your reply! So, what am i supossed to do now?

**Comment by Seldaek** [comment]
The last version of the plugin fixed it so just update the plugin to 2.0.4 and you should be fine https://packagist.org/packages/yiisoft/yii2-composer#2.0.4
